### PR TITLE
Mm 49495 Show past invoices for prior self-hosted purchases

### PR DESCRIPTION
**Pull request by neallred**
#### Summary
Add support code for showing past invoices for prior self-hosted purchases

enterprise PR: https://github.com/mattermost/enterprise/pull/1385

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-49495

#### Release Note
No release note, this mattermost-server PR only supports test code for the backing enterprise code and has a small refactor on a SQL query that doesn't affect anything customer facing.

```release-note
NONE
```

### CODE REVIEW

File: store/sqlstore/license_store.go [hunk 1]
```diff
@@ -33,7 +33,7 @@ func (ls SqlLicenseStore) Save(license *model.LicenseRecord) (*model.LicenseReco
 		return nil, err
 	}
 	query := ls.getQueryBuilder().
-		Select("*").
+		Select("Id, CreateAt, Bytes").
```
Comment (ConorMacpherson) [inline store/sqlstore/license_store.go:36]: 🥇

**Review by ConorMacpherson — APPROVED** [review]

File: model/client4.go [hunk 2]
```diff
@@ -8677,6 +8677,30 @@ func (c *Client4) SelfHostedSignupConfirm(form *SelfHostedConfirmPaymentMethodRe
 	return BuildResponse(r), &response, nil
 }
 
+func (c *Client4) GetSelfHostedInvoices() (*Response, []*Invoice, error) {
```
[the remark this answers is not in the document]
Comment (neallred) [reply model/client4.go:8680]: Yep! The test is in the enterprise PR and depends on this.

**Pull request merged**